### PR TITLE
Feat: RenderCommand factory

### PR DESCRIPTION
**Pull request by Mihinator3000**

### CODE REVIEW

File: Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        return sheetId.Value;
+    }
+
+    private async Task<string> GetSheetTitle(string spreadsheetId, int id, CancellationToken cancellationToken)
+    {
+        IList<Sheet> sheets = await GetSheetsAsync(spreadsheetId, cancellationToken);
+
+        Sheet googleSheet = sheets.FirstOrDefault(s => s.Properties.SheetId == id)
+                            ?? throw new GoogleSheetException($"Sheet with id {id} does not exist");
```
Comment (ronimizy) [inline Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs:59]: Don't really think that null coalescing is the best option for null check here (same with the method above), it is pretty bulky, so I'd rather user just `if (googleSheet is null)...`

**Review by ronimizy — COMMENTED** [review]
please update the README.md

File: Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            throw new GoogleSheetException($"Sheet with title {title} does not exist");
+        }
+
+        int? sheetId = sheet.Properties.SheetId;
+        if (sheetId is null)
+        {
+            throw new GoogleSheetException("Sheet id does not exist");
+        }
+
+        return sheetId.Value;
```
Comment (ronimizy) [inline Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs:53]:
```suggestion
        IList<Sheet> sheets = await GetSheetsAsync(spreadsheetId, cancellationToken);

        Sheet? sheet = sheets.FirstOrDefault(s => s.Properties.Title == title);

        if (sheet is null)
            throw new GoogleSheetException($"Sheet with title {title} does not exist");

        int? sheetId = sheet.Properties.SheetId;

        if (sheetId is null)
            throw new GoogleSheetException("Sheet id does not exist");

        return sheetId.Value;
```

File: Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        IList<Sheet> sheets = await GetSheetsAsync(spreadsheetId, cancellationToken);
+
+        Sheet? sheet = sheets.FirstOrDefault(s => s.Properties.SheetId == id);
+        if (sheet is null)
+        {
+            throw new GoogleSheetException($"Sheet with id {id} does not exist");
+        }
+
+        return sheet.Properties.Title;
```
Comment (ronimizy) [inline Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs:66]:
```suggestion
        IList<Sheet> sheets = await GetSheetsAsync(spreadsheetId, cancellationToken);

        Sheet? sheet = sheets.FirstOrDefault(s => s.Properties.SheetId == id);

        if (sheet is null)
            throw new GoogleSheetException($"Sheet with id {id} does not exist");

        return sheet.Properties.Title; 
```

**Review by ronimizy — APPROVED** [review]

**Comment by Mihinator3000** [comment]
@ronimizy are we using codestyle without additional parentheses when inside block is 1 line?

**Comment by ronimizy** [comment]
> @ronimizy are we using codestyle without additional parentheses when inside block is 1 line?

yep

**Pull request merged**